sample/dir.rb

```diff
@@ -1,12 +1,7 @@
 # directory access
 # list all files but .*/*~/*.o
-dirp = Dir.open(".")
-for f in dirp
-  case f
-  when /\A\./, /~\z/, /\.o\z/
-    # do not print
-  else
-    print f, "\n"
+Dir.open(".").each do |file|
```

Comment:
It is bad style to leave the opened `Dir`.
```suggestion
Dir.foreach(".") do |file|
```